criu/files-reg.c

```diff
@@ -2199,9 +2199,20 @@ int open_path(struct file_desc *d, int (*open_cb)(int mntns_root, struct reg_fil
 		if (!validate_file(tmp, &st, rfi))
 			goto err;
 
-		if (rfi->rfe->has_mode && (st.st_mode != rfi->rfe->mode)) {
-			pr_err("File %s has bad mode 0%o (expect 0%o)\n", rfi->path, (int)st.st_mode, rfi->rfe->mode);
-			goto err;
+		if (rfi->rfe->has_mode) {
+			mode_t curr_mode = st.st_mode;
+			mode_t saved_mode = rfi->rfe->mode;
+
+			if (opts.ignore_rwx_change_on_restore) {
+				curr_mode &= ~(S_IRWXU | S_IRWXG | S_IRWXO);
+				saved_mode &= ~(S_IRWXU | S_IRWXG | S_IRWXO);
+			}
+
+			if (curr_mode != saved_mode) {
+				pr_err("File %s has bad mode 0%o (expect 0%o)\n", rfi->path, (int)curr_mode,
```

Comment:
This error message could be updated to something like:

```
File %s has bad mode 0%o (expect 0%o)
File mode checks can be ignored with the --skip-file-mode-check option
```

Reply:
Done. The failures without the option look as follows:

```
(00.003742) 344968: File root/criu-dev/criu/test/others/skip-file-rwx-check/testfile could only be validated with file size
(00.003747) 344968: Error (criu/files-reg.c:2212): File root/criu-dev/criu/test/others/skip-file-rwx-check/testfile has bad mode 0100444 (expect 0100644)
File r/w/x checks can be skipped with the --skip-file-rwx-check option
(00.003756) 344968: Error (criu/files.c:1213): Unable to open fd=3 id=0x8
(00.003944) Error (criu/cr-restore.c:2536): Restoring FAILED.
```